tests/core/test_processor.py

```diff
@@ -1297,3 +1298,33 @@ def test_predict_next_action_with_hidden_rules():
     action, prediction = processor.predict_next_action(tracker)
     assert isinstance(action, ActionListen)
     assert not prediction.hide_rule_turn
+
+
+def test_predict_next_action_raises_exception(domain: Domain):
```

Comment:
Can we be specific about the exception?

Reply:
Also would be good to test the case where the exception comes from the last action being `action_listen`, not from the limit being reached.

Reply:
> Can we be specific about the exception?
Yup sure, will add the exception name; was afraid of making the line too long.

> test the case where the exception comes from the last action being action_listen
I don't think the exception ever gets triggered when the last action is action_listen, because of the `and` in this [line](https://github.com/RasaHQ/rasa/blob/ac1197b3f80071bb213d4fa66d5d24b4fc01e30e/rasa/core/processor.py#L640) - if last action is action_listen, then `should_predict_another_action` is `False` which will make `is_action_limit_reached` False as well.